### PR TITLE
Print student privacy PDFs

### PR DESCRIPTION
**Pull request by bencodeorg**
# Description

Prints the PDF version of the student privacy page for CSF, CSD, and CSP. Historically, the print button would print something poorly formatted, like the following:

image

Attempted [a version of this ](https://github.com/code-dot-org/code-dot-org/pull/31503) where we would have a general stylesheet for Code.org, but proved too risky. This PR takes a more limited approach and just prints the appropriate PDF for each course.

Although functional (in my local testing), I am certain this could benefit from some improvements. Questions I have:
1. I'm hard coding URL paths here, which is probably not a best practice.
2. I'm using an object to map URLs to PDFs, which might not be good practice in itself (independent of the hard coding).
3. Formatting of my jQuery iframe seems clunky.
4. What kinds of tests can/should I add here?

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/sites/code.org/pages/views/share_privacy.js [hunk 1]
```diff
@@ -25,5 +25,27 @@ $(document).ready(function() {
...
+        '/privacy/student-privacy-csp': '/files/privacy-policy-csp.pdf'
+      };
+
+      let pdfToPrint = pdfToPrintHash[window.location.pathname];
+
+      let iFramePdf = $(
+        '<iframe id="iFramePdf" src=' +
+          pdfToPrint +
+          ' style="display:none;"></iframe>'
+      );
```
Comment (islemaster) [inline apps/src/sites/code.org/pages/views/share_privacy.js]: Could also use a [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)
```js
const iFramePdf = $(`
  <iframe
    id="iFramePdf"
    src="${pdfToPrint}"
  ></iframe>
`);
```

File: apps/src/sites/code.org/pages/views/share_privacy.js [hunk 2]
```diff
@@ -25,5 +25,27 @@ $(document).ready(function() {
...
+      $('body').append(iFramePdf);
+    }
+
+    let getPrintFrame = document.getElementById('iFramePdf');
```
Comment (islemaster) [inline apps/src/sites/code.org/pages/views/share_privacy.js]: Could change to only search the DOM once.
```
let printFrame = document.getElementById('iFramePdf');
if (!printFrame) {
  printFrame = $('...my frame html...')[0];
}
printFrame.focus();
printFrame.contentWindow.print();
```

File: apps/src/sites/code.org/pages/views/share_privacy.js [hunk 3]
```diff
@@ -25,5 +25,27 @@ $(document).ready(function() {
...
       event: 'printed'
     });
+
+    // Prevent repeatedly adding this iframe if button pressed repeatedly
+    if (!$('#iFramePdf').length) {
+      let pdfToPrintHash = {
+        '/privacy/student-privacy': '/files/privacy-policy-csf.pdf',
+        '/privacy/student-privacy-csd': '/files/privacy-policy-csd.pdf',
+        '/privacy/student-privacy-csp': '/files/privacy-policy-csp.pdf'
+      };
```
Comment (islemaster) [inline apps/src/sites/code.org/pages/views/share_privacy.js]: Pull out into a constant at the top of the file. `LOCATION_TO_PDF_MAPPING`

File: apps/src/sites/code.org/pages/views/share_privacy.js [hunk 4]
```diff
@@ -25,5 +25,27 @@ $(document).ready(function() {
...
+        '<iframe id="iFramePdf" src=' +
+          pdfToPrint +
+          ' style="display:none;"></iframe>'
+      );
+      $('body').append(iFramePdf);
+    }
+
+    let getPrintFrame = document.getElementById('iFramePdf');
+    getPrintFrame.focus();
+    getPrintFrame.contentWindow.print();
```
Comment (islemaster) [inline apps/src/sites/code.org/pages/views/share_privacy.js]: To test on different browsers: Does the browser always wait for the PDF to be loaded before trying to print?

**Review by islemaster — COMMENTED** [review]
Looking great! A few style suggestions.

File: apps/src/sites/code.org/pages/views/share_privacy.js (hunk 4 above)
Comment (bencodeorg) [reply apps/src/sites/code.org/pages/views/share_privacy.js]: Tried on Firefox and Safari and ran into a couple problems:
1. If you immediately try to print on page load (in either browser), it prints a blank page.
2. On Firefox, if you wait long enough for it to load, you get `SecurityError: Permission denied to access property "print" on cross-origin object`.

Comment (bencodeorg) [reply apps/src/sites/code.org/pages/views/share_privacy.js]: Welp, this is a bummer:

https://github.com/mozilla/pdf.js/issues/5397

**Comment by bencodeorg** [comment]
Closing this PR for now with blocking issues noted re: printing PDFs using this approach on Firefox.

**Pull request closed without merging**